src/WebJobs.Extensions.DurableTask/LocalHttpListener.cs

```diff
@@ -18,20 +20,36 @@ namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
 {
     internal class LocalHttpListener : IDisposable
     {
+        private const int DefaultPort = 17071;
+
         private readonly DurableTaskExtension extension;
         private readonly IWebHost localWebHost;
         private readonly Func<HttpRequestMessage, Task<HttpResponseMessage>> handler;
 
         public LocalHttpListener(
             DurableTaskExtension extension,
-            Uri listenUri,
             Func<HttpRequestMessage, Task<HttpResponseMessage>> handler)
         {
             this.extension = extension ?? throw new ArgumentNullException(nameof(extension));
             this.handler = handler ?? throw new ArgumentNullException(nameof(handler));
-            this.localWebHost = this.CreateWebHost(listenUri);
+#if !FUNCTIONS_V1
+            this.InternalRpcUri = new Uri($"http://127.0.0.1:{this.GetAvailablePort()}/durabletask/");
+            var listenUri = new Uri(this.InternalRpcUri.GetLeftPart(UriPartial.Authority));
+            this.localWebHost = new WebHostBuilder()
+                .UseKestrel()
+                .UseUrls(listenUri.OriginalString)
+                .Configure(a => a.Run(this.HandleRequestAsync))
+                .Build();
+#else
+            // Just use default port for internal Uri. No need to check for port availability since
+            // we won't be listening to this endpoint.
+            this.InternalRpcUri = new Uri($"http://127.0.0.1:{DefaultPort}/durabletask/");
+            this.localWebHost = new NoOpWebHost();
+#endif
         }
 
+        public Uri InternalRpcUri { get; private set; }
```

Comment:
Looks like we can remove the setter completely since we only assign it in the constructor.